src/modules/Modal/Modal.js

```diff
@@ -133,8 +148,13 @@ class Modal extends Component {
       }
     }
 
-    document.addEventListener('keydown', this.handleDocumentKeyDown)
-    document.addEventListener('click', this.handleClickOutside)
+    if (closeOnEscape) {
+      document.addEventListener('keydown', this.handleDocumentKeyDown)
+    }
```

Comment:
Good call only adding the listeners when needed.  However, if the prop changes after the Modal is open, the listeners will not be added.

We'd need to move these to `componentDidUpdate` and add/remove them based on props.  That logic could get hairy though with lots of conditionals.  I think instead, it is acceptable to allow these two listeners to always exist and we just do nothing in the handlers if the `closeOn*` props are set to false.

Let me know your thoughts.